### DISCUSSION TITLE
Brew zsh hangs on $(...) command substitution in Big Sur 11.0.1

### DISCUSSION THREAD

**Discussion by miiaow** [Homebrew/discussions#141] [Everyday usage] [answered]
Can anyone reproduce this on Big Sur 11.0.1?

install zsh from brew
```
brew install zsh
```

verify version 
```
/usr/local/bin/zsh --version
zsh 5.8 (x86_64-apple-darwin20.1.0)
```

start zsh without rcs (this should start a new shell):
```
/usr/local/bin/zsh --no-rcs
Torgeirs-MacBook-Pro%
```

try to enter the following in the new shell (it should hang):
```
echo $(date)
```

To me it seems like all command substitution using `$()` causes zsh to hang. 
Backticks seem to work though.

**Comment by Aljutor** [comment]
Yes, I have same problem, zsh 5.8, Big Sur 11.0.1

**Comment by johndbritton** [comment]
> Can anyone reproduce this on Big Sur 11.0.1?

Yes, I have the same issue. For the time being I've switched my default shell to the version of `zsh` provided with macOS.

In case others end up on this thread via Google like me, here's a temporary workaround:

Open `Terminal.app` preferences and set `Shells open with` to `Command (complete path):` and use `/bin/zsh` for the macOS provided `zsh` binary.

Now when you open `Terminal.app` `zsh` will no longer hang. If you want to update your default user shell you can use `sudo dscl . -create /Users/$USER UserShell /bin/zsh`.

**Comment by miiaow** [comment]
Issue created: https://github.com/Homebrew/homebrew-core/issues/64921

**Comment by fxcoudert** [comment, marked as answer]
I've identified it as a zsh bug with clang 12, and reported it to the zsh developers.
A workaround is being put in place for Homebrew: https://github.com/Homebrew/homebrew-core/pull/64923

**Comment by carlocab** [comment]
This has now been fixed. Quoting the second part of [this comment](https://github.com/Homebrew/homebrew-core/issues/64921#issuecomment-727925770):

>One hour from report to analysis, upstream report, and local workaround is not bad for a volunteer-based team. Please consider sponsoring our work if you can: https://github.com/sponsors/Homebrew

Thanks for getting this done so quickly, @fxcoudert !
